app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -444,7 +444,7 @@ public function handleTransactionUpdate(
         $newTransactionStatus,
         $prevTransactionStatus,
         $transactionAmount = null,
-        $transaction = null
```

Comment:
This PR changes behaviour when handleTransactionUpdate is called w/o $transaction (eg., https://github.com/BoltApp/bolt-magento1/blob/2371d2b4e1633411962bf905a440c01f51a951c4/app/code/community/Bolt/Boltpay/Model/Payment.php#L422) 

Should we fetch transaction if caller is not passing it?

Reply:
@daisy1754 
The above call here is ok.  The behavior will be the same for placing a transaction on hold.  Your idea is full proof however and I will add this.



`handleTransactionUpdate` does not fetch the transaction.  If it is called without the transaction, no fetching will occur.  

The caller of `handleTransactionUpdate`  is responsible for deciding if fetching the transaction from bolt is necessary.  This method is normally called from the webhook controller.

Reply:
done